src/client/datascience/interactive-common/interactiveBase.ts

```diff
@@ -127,7 +126,8 @@ export abstract class InteractiveBase extends WebviewPanelHost<IInteractiveWindo
     }
 
     public abstract isInteractive: boolean;
-    protected abstract get notebookMetadata(): INotebookMetadataLive | undefined;
+    protected abstract get notebookMetadata(): Readonly<nbformat.INotebookMetadata> | undefined;
+    protected abstract get kernelConnection(): Readonly<KernelConnectionMetadata> | undefined;
```

Comment:
Converted these to readonly so that we can ensure we never attempt to update any properties.
Not that we have, e.g. in the case of notebook metadata there's a very specific way to update the metadata (you cannot edito this directly).
Same way with connections, added comments in the code why they are readonly (future proofing & ensure they are treated as immutable types).